### ISSUE TITLE
[BUG] Galera Cluster mysql 5.7.21 normal shutdown

### ISSUE THREAD

**Issue by joaoleal90** [mariadb-corporation/mysql-wsrep#342]
Every time i try to stop the node gives this error:

pure virtual method called
terminate called without an active exception
12:49:26 UTC - mysqld got signal 6 ;
This could be because you hit a bug. It is also possible that this binary
or one of the libraries it was linked against is corrupt, improperly built,
or misconfigured. This error can also be caused by malfunctioning hardware.
Attempting to collect some information that could help diagnose the problem.
As this is a crash and something is definitely wrong, the information
collection process might fail.

key_buffer_size=10737418240
read_buffer_size=131072
max_used_connections=14
max_threads=1500
thread_count=1
connection_count=0
It is possible that mysqld could use up to
key_buffer_size + (read_buffer_size + sort_buffer_size)*max_threads = 11082642 K  bytes of memory
Hope that's ok; if not, decrease some variables in the equation.

Thread pointer: 0x0
Attempting backtrace. You can use the following information to find out
where mysqld died. If you see no messages after this, something went
terribly wrong...
stack_bottom = 0 thread_stack 0x40000
/usr/sbin/mysqld(my_print_stacktrace+0x35)[0xf7ab45]
/usr/sbin/mysqld(handle_fatal_signal+0x4a4)[0x7d8354]
/lib64/libpthread.so.0[0x306040f7e0]
/lib64/libc.so.6(gsignal+0x35)[0x3060032495]
/lib64/libc.so.6(abort+0x175)[0x3060033c75]
/usr/lib64/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x12d)[0x30644bea8d]
/usr/lib64/libstdc++.so.6[0x30644bcbe6]
/usr/lib64/libstdc++.so.6[0x30644bcc13]
/usr/lib64/libstdc++.so.6[0x30644bd56f]
/usr/sbin/mysqld(_ZN18Global_THD_manager14do_for_all_thdEP11Do_THD_Impl+0x90)[0x7d6570]
/usr/sbin/mysqld(_Z25wsrep_wait_appliers_closeP3THD+0x75)[0x7c8c95]
/usr/sbin/mysqld(_Z22wsrep_stop_replicationP3THD+0xb5)[0x7e0ce5]
/usr/sbin/mysqld(signal_hand+0x628)[0x7cc318]
/usr/sbin/mysqld(pfs_spawn_thread+0x174)[0x123ba34]
/lib64/libpthread.so.0[0x3060407aa1]
/lib64/libc.so.6(clone+0x6d)[0x30600e8bdd]
The manual page at http://dev.mysql.com/doc/mysql/en/crashing.html contains
information that should help you find out what is causing the crash.

Config node 1:
#CONFIG MYSQL+GALERA
[mysqld]
user=mysql
binlog_format=ROW
explicit_defaults_for_timestamp = 1
max_connections = 1500
bind-address =0.0.0.0

#InnoDB
default_storage_engine=innodb
innodb_file_per_table
innodb_flush_method=O_DIRECT
innodb_autoinc_lock_mode=2
innodb_flush_log_at_trx_commit=0
innodb_buffer_pool_size=30G
innodb_buffer_pool_instances=15
innodb_log_file_size=500M
innodb_log_files_in_group=3
key_buffer_size=10G
innodb_force_recovery =0
#LOGS
general_log_file=/var/lib/mysql/logs/mysql_general.log
general_log=1
log_error=/var/lib/mysql/logs/mysqld.log
#log-slow-admin-statements=ON
#long_query_time=2
#log-slow-queries=/var/lib/mysql/logs/mysqld_slow.log

#SQL_QUERYS
sql_mode = "STRICT_TRANS_TABLES,NO_ZERO_IN_DATE,NO_ZERO_DATE,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION"
query_cache_size=0
query_cache_type=0

#[galera]
#WSREP
wsrep_on=ON
wsrep_provider=/usr/lib64/galera-3/libgalera_smm.so
wsrep_provider_options="gcache.size=300M; gcache.page_size=300M"
wsrep_cluster_name="PRDDB"
wsrep_cluster_address="gcomm://10.239.62.11:4567,10.239.62.12:4567,10.239.62.13:4567"
wsrep_sst_method=rsync
wsrep_node_address="10.239.62.11"
wsrep_node_name="GADB01PRD"

How to repeat:
service mysqld stop

2018-09-20T12:49:24.979284Z 0 [Note] WSREP: Stop replication
2018-09-20T12:49:24.979336Z 0 [Note] WSREP: Closing send monitor...
2018-09-20T12:49:24.979344Z 0 [Note] WSREP: Closed send monitor.
2018-09-20T12:49:24.979431Z 0 [Note] WSREP: gcomm: terminating thread
2018-09-20T12:49:24.979459Z 0 [Note] WSREP: gcomm: joining thread
2018-09-20T12:49:24.979516Z 0 [Note] WSREP: gcomm: closing backend
2018-09-20T12:49:24.980339Z 0 [Note] WSREP: view(view_id(NON_PRIM,305860c1,124) memb {
        b25fc07f,0
} joined {
} left {
} partitioned {
        305860c1,0
        d0e5ef58,0
})
2018-09-20T12:49:24.980387Z 0 [Note] WSREP: view((empty))
2018-09-20T12:49:24.980592Z 0 [Note] WSREP: gcomm: closed
2018-09-20T12:49:24.980621Z 0 [Note] WSREP: New COMPONENT: primary = no, bootstrap = no, my_idx = 0, memb_num = 1
2018-09-20T12:49:24.980654Z 0 [Note] WSREP: Flow-control interval: [16, 16]
2018-09-20T12:49:24.980664Z 0 [Note] WSREP: Trying to continue unpaused monitor
2018-09-20T12:49:24.980669Z 0 [Note] WSREP: Received NON-PRIMARY.
2018-09-20T12:49:24.980674Z 0 [Note] WSREP: Shifting SYNCED -> OPEN (TO: 103795)
2018-09-20T12:49:24.980692Z 0 [Note] WSREP: Received self-leave message.
2018-09-20T12:49:24.980701Z 0 [Note] WSREP: Flow-control interval: [0, 0]
2018-09-20T12:49:24.980706Z 0 [Note] WSREP: Trying to continue unpaused monitor
2018-09-20T12:49:24.980710Z 0 [Note] WSREP: Received SELF-LEAVE. Closing connection.
2018-09-20T12:49:24.980715Z 0 [Note] WSREP: Shifting OPEN -> CLOSED (TO: 103795)
2018-09-20T12:49:24.980725Z 0 [Note] WSREP: RECV thread exiting 0: Success
2018-09-20T12:49:24.980724Z 2 [Note] WSREP: New cluster view: global state: 606884f7-b77a-11e8-a97b-4f772a6f95af:103795, view# -1: non-Primary, number of nodes: 1, my index: 0, protocol version 3
2018-09-20T12:49:24.980771Z 2 [Note] WSREP: wsrep_notify_cmd is not defined, skipping notification.
2018-09-20T12:49:24.980750Z 0 [Note] WSREP: recv_thread() joined.
2018-09-20T12:49:24.980804Z 0 [Note] WSREP: Closing replication queue.
2018-09-20T12:49:24.980832Z 0 [Note] WSREP: Closing slave action queue.
2018-09-20T12:49:24.980844Z 0 [Note] Giving 15 client threads a chance to die gracefully
2018-09-20T12:49:24.980813Z 2 [Note] WSREP: New cluster view: global state: 606884f7-b77a-11e8-a97b-4f772a6f95af:103795, view# -1: non-Primary, number of nodes: 0, my index: -1, protocol version 3
2018-09-20T12:49:24.980875Z 2 [Note] WSREP: wsrep_notify_cmd is not defined, skipping notification.
2018-09-20T12:49:24.980862Z 0 [Warning] /usr/sbin/mysqld: Forcing close of thread 125  user: 'root'

**2018-09-20T12:49:24.980899Z 2 [Note] WSREP: applier thread exiting (code:0

pure virtual method called
terminate called without an active exception
12:49:26 UTC - mysqld got signal 6 ;
This could be because you hit a bug. It is also possible that this binary
or one of the libraries it was linked against is corrupt, improperly built,
or misconfigured. This error can also be caused by malfunctioning hardware.
Attempting to collect some information that could help diagnose the problem.
As this is a crash and something is definitely wrong, the information
collection process might fail.

key_buffer_size=10737418240
read_buffer_size=131072
max_used_connections=14
max_threads=1500
thread_count=1
connection_count=0
It is possible that mysqld could use up to
key_buffer_size + (read_buffer_size + sort_buffer_size)*max_threads = 11082642 K  bytes of memory
Hope that's ok; if not, decrease some variables in the equation.

Thread pointer: 0x0
Attempting backtrace. You can use the following information to find out
where mysqld died. If you see no messages after this, something went
terribly wrong...
stack_bottom = 0 thread_stack 0x40000
/usr/sbin/mysqld(my_print_stacktrace+0x35)[0xf7ab45]
/usr/sbin/mysqld(handle_fatal_signal+0x4a4)[0x7d8354]
/lib64/libpthread.so.0[0x306040f7e0]
/lib64/libc.so.6(gsignal+0x35)[0x3060032495]
/lib64/libc.so.6(abort+0x175)[0x3060033c75]
/usr/lib64/libstdc++.so.6(_ZN9__gnu_cxx27__verbose_terminate_handlerEv+0x12d)[0x30644bea8d]
/usr/lib64/libstdc++.so.6[0x30644bcbe6]
/usr/lib64/libstdc++.so.6[0x30644bcc13]
/usr/lib64/libstdc++.so.6[0x30644bd56f]
/usr/sbin/mysqld(_ZN18Global_THD_manager14do_for_all_thdEP11Do_THD_Impl+0x90)[0x7d6570]
/usr/sbin/mysqld(_Z25wsrep_wait_appliers_closeP3THD+0x75)[0x7c8c95]
/usr/sbin/mysqld(_Z22wsrep_stop_replicationP3THD+0xb5)[0x7e0ce5]
/usr/sbin/mysqld(signal_hand+0x628)[0x7cc318]
/usr/sbin/mysqld(pfs_spawn_thread+0x174)[0x123ba34]
/lib64/libpthread.so.0[0x3060407aa1]
/lib64/libc.so.6(clone+0x6d)[0x30600e8bdd]
The manual page at http://dev.mysql.com/doc/mysql/en/crashing.html contains
information that should help you find out what is causing the crash.**

**Comment by temeo** [comment]
Maybe related crash with debug build:
```
2018-12-15T10:27:51.948700Z 6 [Note] WSREP: Shifting CLOSED -> DESTROYED (TO: 8)
2018-12-15T10:27:54.164783Z 6 [Note] WSREP: Flushing memory map to disk...
terminate called after throwing an instance of 'gu::Exception'
  what():  Mutex lock failed: Invalid argument
	 at galera/src/gcs_action_source.cpp:dispatch():138
	 at galera/src/gcs_action_source.cpp:process():180
10:27:55 UTC - mysqld got signal 6 ;
```

**Comment by joaoleal90** [comment]
I Keep having problens every time i do run "service mysqld stop"

**Comment by joaoleal90** [comment]
Repos:

[galera]
name = Galera
baseurl = http://releases.galeracluster.com/galera-3/redhat/6/x86_64/
gpgkey = http://releases.galeracluster.com/GPG-KEY-galeracluster.com
gpgcheck = 1

[mysql-wsrep]
name = MySQL-wsrep
#baseurl =  http://releases.galeracluster.com/mysql-wsrep-5.7/redhat/6/x86_64/
baseurl =  http://releases.galeracluster.com/mysql-wsrep-5.7.21-25.14/redhat/6/x86_64/
gpgkey = http://releases.galeracluster.com/GPG-KEY-galeracluster.com
gpgcheck = 1

**Comment by temeo** [comment]
Hi, this should be fixed in upcoming 5.7.26-25.18 (was duplicate of https://github.com/codership/mysql-wsrep/issues/298).
